client/network/src/protocol/generic_proto/handler.rs

```diff
@@ -971,6 +971,7 @@ impl ProtocolsHandler for NotifsHandler {
 						if let Some(pos) = self.out_protocols.iter().position(|(n, _)| *n == protocol_name) {
 							if let Some(substream) = out_substreams[pos].as_mut() {
 								let _ = substream.start_send_unpin(message);
+								cx.waker().wake_by_ref();
```

Comment:
Would you mind documenting that this wake up is used to force another call to `NotifsHandler::poll` and thus to flush the corresponding substreams?

In addition I would suggest documenting that this is temporary and removed with https://github.com/paritytech/substrate/pull/7374.